### DISCUSSION TITLE
I can't download playlists at all and I can't see how to enable the feature

### DISCUSSION THREAD

**Discussion by Nikko-A** [NickvisionApps/Parabolic#121] [Q&A]
Hello! I'm on the newest version of Tube Converter from flathub and I'm currently having issues on trying to download playlists. With other similar programs the playlist download feature is separated, but I can't see it here. I assumed it was because the "Video URL" serves as the box that I put the playlist URL too but it returns with an error. I assume this is just user error, wasn't able to find documentation for playlist download yet, so I would like to ask for help please.

**Comment by nlogozzo** [comment]
What is the link of the playlist you are trying to download? I'll see if it fails on my end.

**Comment by fsobolev** [comment]
> I assumed it was because the "Video URL" serves as the box that I put the playlist URL too but it returns with an error. I assume this is just user error, wasn't able to find documentation for playlist download yet, so I would like to ask for help please.

You did correctly when pasted playlist url in the url entry, it is supposed to work. Tube Converter uses yt-dlp under the hood, so everything that yt-dlp accepts should not produce an error in Tube Converter. In case of youtube, both `https://www.youtube.com/playlist?list=PlaylistID` and `https://www.youtube.com/watch?v=VideoID&list=PlaylistID` are valid url formats to download a playlist.
You can try to see if yt-dlp understands the link you tried to use:
```
yt-dlp --dump-json "URL"
```
In case of failure, it will show you an error, otherwise there will be a giant json that shows that yt-dlp is able to retrieve the data.
If you don't have yt-dlp installed on the host, you can use the one inside Tube Converter flatpak:
```
flatpak run --command=sh org.nickvision.tubeconverter
```
And then run the yt-dlp command.

**Comment by nlogozzo** [comment]
I was actually able to reproduce this: 
I tried downloading the playlist: https://www.youtube.com/watch?v=q5q1Ec01W9M&list=PL4082F63E4EE11B48
And I get the error 
```
"ERROR: [youtube] KaVnA88oNTg: Private video. Sign in if you've been granted access to this video"
```
It turns out there is just 1 video unavailable in the playlist, but it causes TC to say the whole playlist is invalid.

**Reply by nlogozzo** [reply]
See #124
